DSpace/dspace-angular
# Fix missing hints and error messages for input-type `list` & `tag`

## References
* Fixes #1770
* Fixes #2131

## Description
Fixed a problem causing the hints for input types `list` and `tag` to always be hidden.
Fixed some problems causing the error messages and the asterisk not being displayed for input type `list`.

## Instructions for Reviewers

List of changes in this PR:
* Input types which had array values (`tag` & `list`) could not display their hint property when they were marked as `<repeatable>true</repeatable>`. This was because repeatable fields normally show their hints not underneath each field but in a section underneath all the input fields of that group with the `+ Add more` button, but for those 2 types of input types that whole section to add more input fields is hidden. So I added an exception in `DsDynamicFormControlContainerComponent` to also display the hint under repeatable dynamic array fields.
* The required and hint property where also not serialised
* The field validation for repeatable fields with input type `list` also checked if every checkbox was selected individually instead of checking if at least one was selected.

**Guidance for how to test or review your PR:**
* Use the default `submission-forms.xml` ➜ check that the hint of the `dc.subject` field is visible
* Set the `input-type` of `dc.type` & `dc.language.iso` to `list`
  * and provide an error message in the `required` tag:
    * Check that the asterisk is visible
    * Check that the error message is visible when you submit the item without filling out the fields
    * Check that the error message disappears when a value is selected and that you can submit the item
  * and leave the `required` tag empty:
    * Check that no asterisk is shown
    * Check that no error message is shown when submitting the item

## Checklist
